app/js/services/services.js

```diff
@@ -587,6 +623,120 @@ KMCServices.factory('PlayerService', ['$http', '$modal', '$log', '$q', 'apiServi
+		}
+
+		function cleanupV7PlayerConfig(playerConfig) {
+			playerConfig['uiConf:confFile'] = ' ';
+			playerConfig['uiConf:swfUrlVersion'] = '';
+			playerConfig['uiConf:html5Url'] = '';
+
+			return playerConfig;
+		}
+
+		var playerDataService = {
+			getDefaultV7PlayerConf: function (name, description) {
+				var deferred = $q.defer();
+				var result = {
+					'uiConf:objectType': 'KalturaUiConf',
+					'uiConf:objType': 1,
+					'uiConf:description': description || '',
+					'uiConf:height': 395,
+					'uiConf:width': 560,
+					'uiConf:swfUrl': '/',
+					'uiConf:name': name,
+					'uiConf:tags': 'kalturaPlayerJs,player,ovp',
+					'uiConf:config': '{\n' +
+						'    "disableUserCache": false,\n' +
+						'    "plugins": {\n' +
+						'        "kava": {},\n' +
+						'        "playkit-js-hotspots": {},\n' +
+						'        "navigation": {},\n' +
+						'        "playkit-js-transcript": {\n' +
+						'            "position": "bottom",\n' +
+						'            "showTime": true,\n' +
+						'            "expandOnFirstPlay": false,\n' +
+						'            "downloadDisabled": true,\n' +
+						'            "printDisabled": true\n' +
+						'        },\n' +
+						'        "playkit-js-info": {},\n' +
+						'        "dualscreen": {},' +
+						'        "timeline": {},\n' +
+						'        "ivq": {},\n' +
+						'        "kalturaCuepoints": {}' +
+						'    },\n' +
+						'    "viewability": {\n' +
+						'        "playerThreshold": 50\n' +
+						'    },\n' +
+						'    "provider": {\n' +
+						'        "env": {}\n' +
+						'    },\n' +
+						'    "playback": {\n' +
+						'        "enabled": true\n' +
+						'    }\n' +
+						'}',
+					'uiConf:confVars': '{"versions":{"kaltura-ovp-player":"{latest}","playkit-hotspots":"{latest}",' +
+						'"playkit-navigation":"{latest}","playkit-transcript":"{latest}","playkit-info":"{latest}",' +
+						'"playkit-ivq":"{latest}","playkit-timeline":"{latest}","playkit-kaltura-cuepoints":"{latest}",' +
+						'"playkit-dual-screen":"{latest}"},"langs":["en"]}',
+					'uiConf:creationMode': 2,
+				};
+
+				result = cleanupV7PlayerConfig(result);
+
+				deferred.resolve(result);
+				return deferred.promise;
+			},
+
+			getV7PlayerConfFromTemplate: function (name, playerId) {
+				var deferred = $q.defer();
+				apiService.setCache(false);
+				var request = {
+					'service': 'uiConf',
+					'action': 'get',
+					'id': playerId
+				};
+				apiService.doRequest(request).then(function (result) {
+						if (!validateV7Player(result)) {
+							deferred.reject($filter('translate')('The provided player ID is not a V7 player.\nPlayer ID: ') + result.id);
+							return;
+						}
+						// validate result to catch invalid JSON configs
+						if (typeof result.config === 'string') {
+							try {
+								angular.fromJson(result.config);
+								var ignoredFields = ['id', 'name', 'description', 'partnerId', 'objTypeAsString', 'createdAt', 'updatedAt', 'version'];
```

Comment:
@vladimir-poleh  did you compare the list of attributes you want to take?

Reply:
I'm assuming that v7 player we are using to copy from is valid and contains all necessary properties. Ignored fields are read only fields that can't be updated.

Default v7 player contains only v7 player properties, old v2 properties that are not used in v7 will be wiped during upgrade for both upgrade to default and upgrade to pre-defined player.